qcodes/station.py

```diff
@@ -217,3 +258,245 @@ def __getitem__(self, key):
+
+        Loading of a configuration will update the snapshot of the station and
+        make the instruments described in the config file available for
+        instantiation with the :meth:`load_instrument` method.
+
+        Additionally the shortcut methods `load_<instrument_name>` will be
+        updated.
+        """
+        def update_station_configuration_snapshot():
+            class StationConfig(UserDict):
+                def snapshot(self, update=True):
+                    return self
+
+            self.components['config'] = StationConfig(self._config)
+
+        def update_load_instrument_methods():
+            #  create shortcut methods to instantiate instruments via
+            # `load_<instrument_name>()` so that autocompletion can be used
+            # first remove methods that have been added by a previous
+            # :meth:`load_config_file` call
+            while len(self._added_methods):
+                delattr(self, self._added_methods.pop())
+
+            # add shortcut methods
+            for instrument_name in self._instrument_config.keys():
+                method_name = f'load_{instrument_name}'
+                if method_name.isidentifier():
+                    setattr(self, method_name, partial(
+                        self.load_instrument, identifier=instrument_name))
+                    self._added_methods.append(method_name)
+                else:
+                    log.warning(f'Invalid identifier: ' +
+                                f'for the instrument {instrument_name} no ' +
+                                f'lazy loading method {method_name} could ' +
+                                'be created in the Station.')
+        self._config = YAML().load(config)
+        self._instrument_config = self._config['instruments']
+        update_station_configuration_snapshot()
+        update_load_instrument_methods()
+
+    def close_and_remove_instrument(self,
+                                    instrument: Union[Instrument, str]
+                                    ) -> None:
+        """
+        Safely close instrument and remove from station and monitor list
+        """
+        # remove parameters related to this instrument from the
+        # monitor list
+        if isinstance(instrument, str):
+            instrument = Instrument.find_instrument(instrument)
+
+        self._monitor_parameters = [v for v in self._monitor_parameters
+                                    if v.root_instrument is not instrument]
+        # remove instrument from station snapshot
+        self.remove_component(instrument.name)
+        # del will remove weakref and close the instrument
+        instrument.close()
+        del instrument
+
+    def load_instrument(self, identifier: str,
+                        revive_instance: bool = False,
+                        **kwargs) -> Instrument:
+        """
+        Creates an :class:`~qcodes.Instrument` instance as described by the
```

Comment:
I think you should either use the full path to the actual Class or just simply  `.Instrument` which should resolve to the correct one

Reply:
I was also unsure about it, so I tested it locally and it worked. As CI is failing I will update it to the full path.